### ISSUE TITLE
[Feature request] Markdown syntax, which user is preferable, for GitHub issue creating

### ISSUE THREAD

**Issue by Kristinita** [divinites/gissues#50]
### Request

Add by default the possibility to select for creating new issue syntax that the user typically uses for Markdown documents.
### Justification
1. In `MarkdownEditing — Markdown GFM` syntax I have many snippets, completions, macros and syntax highlight for computer languages when I use only in Markdown GFM syntax. When I do not use snippets, completions, macros and syntax highlight for computer languages I spend a lot of time.
2. Since GitHub Issues use Markdown markup, I see no point in creating an additional syntax, if we have syntaxes for Markdown.
### Extended behavior

`GitHub Issue: Create Issue` → I select user/repository → GitHub Issue file always was opened in `MarkdownEditing — Markdown GFM` syntax (or other syntax, which users use for Markdown documents), not `GitHub Issue — Markdown` syntax. 
### Not helped

I can not always use for issues crating MarkdownEditing — Markdown GFM syntax because file, when I wrote issue, not have extension. I can not add my preferable Markdown syntax for create issue file:
1. `View` → `Syntax` → `Open all with current extension as...` → `MarkdownEditing` → `Markdown GFM`.
2. [ApplySyntax](https://github.com/facelessuser/ApplySyntax) plugin.
### Environment

**Operating system and version:**
Windows 10.0.14393
**Sublime Text:**
Build 3126

Thanks.

**Comment by divinites** [comment]
@kristinita,  Thanks for the feature request. As far as I know, Sublime Text have a very convenient way to change current view's syntax.

In a Windows env, you can type <kbd>ctrl + shift + p </kbd> to open command palette, and type `sstx markdown`, and you will be able to see all Markdown related syntax.

Maybe you have paid attention to the fact that I have changed the default issue syntax from MarkdownGFM to Sublime Text's Default Markdown, I did this for three reasons:
- The default Markdown Syntax is error-free, no annoying errors like 

```
Error loading scope:source.css, source.less: Unable to find syntax file for scope source.css, source.less in Packages/GitHubIssue/issue.tmLanguage
Error loading scope:source.coffee: Unable to find syntax file for scope source.coffee in Packages/GitHubIssue/issue.tmLanguage
Error loading scope:source.scheme: Unable to find syntax file for scope source.scheme in Packages/GitHubIssue/issue.tmLanguage
Error loading scope:source.css, source.sass: Unable to find syntax file for scope source.css, source.sass in Packages/GitHubIssue/issue.tmLanguage
Error loading scope:source.swift: Unable to find syntax file for scope source.swift in Packages/GitHubIssue/issue.tmLanguage
rule scope:source.scala#xml-comments has a scope name, but is unreachable, so the name will never be used
```
- The default markdown syntax adopted the new format `.sublime-syntax` instead of `.tmLanguage` which is actually TextMate syntax definition.
- Since the plug-in use <kbd>ctrl+s</kbd> and <kbd>ctrl+shift+s</kbd> as the default short-cut of `github_update_issue` and `github_post_issue`, to avoid conflicting with the system default behaviour (<kbd>ctrl+s</kbd> for quick save, it is a very important shortcut system-wide!) I need to have a specific syntax definition and impose a constraint (scope selector) to make sure that these shortcuts are only valid in this specific syntax definition. 

To sum up, there have to be a default syntax (here is issue.sublime-syntax, a modified version of system Markdown syntax) for issues and comments to avoid shortcut conflicts. But you do have the freedom to override the syntax using `command palette`. Once you switch to another syntax, the shortcuts will no more work and you will have to run all commands through `command palette` too.

**Comment by divinites** [comment]
Actually, because of the new regex engine adopted by the dev version of Sublime Text, I found most of the Markdown Syntax definition plug-in, as are listed in [this page](https://packagecontrol.io/search/markdown) are more or less problematic, and generate similar errors as I posted above. If you found an error-free syntax definition, please do let me know, since I am also eager to use extended markdown syntax other than the system default.

**Comment by Kristinita** [comment]
> In a Windows env, you can type ctrl + shift + p to open command palette, and type sstx markdown, and you will be able to see all Markdown related syntax. 
1. I need to change the syntax every time I create a new issue. It is essential that the user had enough change syntax only one time.
2. Before I post or update issue I back need change my syntax as Markdown GFM to Markdown.

Thanks.

**Comment by divinites** [comment]
Okay, no problem,  I can add a configure parameter in sublime-settings, but you will lose the short-cut keys after changing the syntax.

**Comment by Kristinita** [comment]
When I wrote previous comment, I get stack trace:

``` python
command: update_github_issue {}
Traceback (most recent call last):
  File "./python3.3/logging/__init__.py", line 941, in emit
  File "./python3.3/encodings/cp1251.py", line 19, in encode
UnicodeEncodeError: 'charmap' codec can't encode character '\u2192' in position 798: character maps to <undefined>
Logged from file __init__.py, line 1794
No git :( [Errno 2] No such file or directory: 'E:\\Киролайна\\.git\\HEAD'
```

Thanks.

**Comment by divinites** [comment]
That's interesting, since \u2192 is this  "→", and it seems that it happens when python try to logging... put debug flag to 0 might solve this problem then?

**Comment by divinites** [comment]
solved by e16d7dd.

**Comment by divinites** [comment]
``` json
{
    "token": "xxx",
    "username": "divinites",
    "debug": 1,
    "syntax": "Packages/Markdown/MultiMarkdown.sublime-syntax",
    "git_path": ""

}
```

To customize syntax and solve "no git" problem, I add two parameters: _syntax_ and _git_path_. You will need to fill in proper value into it.

### how to obtain syntax path:

Open a markdown file -> set the syntax you want -> open command prompt ->type `view.settings().get('syntax')` -> copy and paste the syntax path to the configure file.

**Comment by Kristinita** [comment]
@divinites , again `CR` on end of lines for `MarkdownEditing — MarkdownGFM` syntax:

![CR1](http://i.imgur.com/8kL7y8o.png)
![CR2](http://i.imgur.com/n4VOHf7.png)

**Comment by divinites** [comment]
CR problem refers to #49 , if you can use other syntax, this issue is solved then.

**Comment by divinites** [comment]
@Kristinita I have elegantly solved #50. Acutally I have completely removed `issue.sublime-syntax`, and use `Packages/Markdown/Markdown.sublime-syntax` as the default value. Of course, you can use whatever markdown syntax as you wish, the most important thing is that now under any custom syntax file, you are able to use all shortcut keys and auto-completions! Also, there is no need to set custom_scope, the new `github_issue.sublime-settings` is:

``` json
{
    "token": "",
    "username": "",
    "password": "",
    "debug": 0,
    "syntax": "Packages/Markdown/Markdown.sublime-syntax",
    "git_path": "",
    "issue_title_completion": true,
    "user_completion": true,
    "label_completion": true,
    "commit_completion": true,
    "commit_completion_trigger": "&"
}
```

Following your recommendation, I also installed Github Emoji, but I found when I type "@", the emojis also pop up with names/labels.
